module/documents/actor/actor.mjs

```diff
@@ -1105,6 +1133,88 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
   /* -------------------------------------------- */
 
+  /**
+   * Initiate concentration on an item.
+   * @param {Item5e} item                     The item on which to being concentration.
+   * @param {object} [effectData]             Effect data to merge into the created effect.
+   * @returns {Promise<ActiveEffect5e[]>}     A promise that resolves to the created and deleted effects.
+   */
+  async beginConcentratingOn(item, effectData={}) {
+    effectData = ActiveEffect5e.createConcentrationEffectData(item, effectData);
+
+    /**
+     * A hook that is called before a concentration effect is created.
+     * @function dnd5e.preBeginConcentrating
+     * @memberof hookEvents
+     * @param {Actor5e} actor         The actor initiating concentration.
+     * @param {Item5e} item           The item that will be concentrated on.
+     * @param {object} effectData     Data used to create the ActiveEffect.
+     * @returns {boolean}             Explicitly return false to prevent the effect from being created.
+     */
+    if ( Hooks.call("dnd5e.preBeginConcentrating", this, item, effectData) === false ) return;
+
+    const effect = await ActiveEffect5e.create(effectData, { parent: this });
+
+    /**
+     * A hook that is called after a concentration effect is created.
+     * @function dnd5e.createConcentrating
+     * @memberof hookEvents
+     * @param {Actor5e} actor             The actor initiating concentration.
+     * @param {Item5e} item               The item that is being concentrated on.
+     * @param {ActiveEffect5e} effect     The created ActiveEffect instance.
+     */
+    Hooks.callAll("dnd5e.beginConcentrating", this, item, effect);
+
+    return effect;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * End concentration on an item.
+   * @param {Item5e|ActiveEffect5e|string} target     An item or effect to end concentration on, or id of an effect.
+   * @returns {Promise<ActiveEffect5e>}               A promise that resolves to the deleted effect.
```

Comment:
```suggestion
   * @returns {Promise<ActiveEffect5e|void>}               A promise that resolves to the deleted effect.
```

Reply:
I changed this into always returning an array instead. I think that makes a lot of sense.